src/main/java/io/jenkins/plugins/gitlabbranchsource/GitLabHookCreator.java

```diff
@@ -145,13 +145,15 @@ public static void createSystemHookWhenMissing(GitLabServer server,
         }
     }
 
-    public static String getHookUrl(boolean isWebHook) {
+    public static String getHookUrl(boolean isWebHook, boolean hostedOnOwnServer) {
         JenkinsLocationConfiguration locationConfiguration = JenkinsLocationConfiguration.get();
         String rootUrl = locationConfiguration.getUrl();
         if (StringUtils.isBlank(rootUrl)) {
             return "";
         }
-        checkURL(rootUrl);
+        if (!hostedOnOwnServer) {
```

Comment:
What is this trying to check?
hosted on own server tells me nothing.

Are you talking about hosting gitlab and jenkins on the same server?
Or is it GitLab.com vs gitliab on-prem?

Please disable webhooks if your using localhost.
GitLab API does not allow registering webhook in any case with localhost or a non fully qualified domain name.

Reply:
See screenshot in this issue https://issues.jenkins-ci.org/browse/JENKINS-59759?filter=-2. Self hosted GitLab can be configured to setup hooks on localhost.

Reply:
Just because it can. doesn't mean it's a good idea 😭